### PR TITLE
use isObject

### PR DESCRIPTION
**Pull request by basz**
#51

### CODE REVIEW

File: src/utils/set-deep.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default function setDeep(
     if (!obj) {
       options.safeSet(target, prop, {});
     } else if (obj && target[prop] instanceof Change) {
-      if (typeof target[prop].value === 'object') {
+      if (isObject(target[prop].value)) {
```
Comment (snewcomer) [inline src/utils/set-deep.ts:69]: Looks like we might need an import?

Comment (basz) [reply src/utils/set-deep.ts:69]: line 65? we got it alredy

Comment (basz) [reply src/utils/set-deep.ts:69]: image

Comment (basz) [reply src/utils/set-deep.ts:69]: oh. you mean we should use the one from is-object.ts?

**Comment by basz** [comment]
I've changed the import. Please please check if that is what you meant as the import does not check typeof x === 'function'

File: src/utils/set-deep.ts [hunk 2]
```diff
@@ -26,10 +27,6 @@ function isValidKey(key: unknown) {
   return key !== '__proto__' && key !== 'constructor' && key !== 'prototype';
 }
 
-function isObject(val: unknown) {
-  return val !== null && (typeof val === 'object' || typeof val === 'function');
-}
```
Comment (snewcomer) [inline src/utils/set-deep.ts:31]: Yep!  Not sure why we had the function check.  As long as all the tests pass!  (Don't have CI setup yet)

**Pull request merged**